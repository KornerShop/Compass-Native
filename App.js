import React from 'react'
import {StyleSheet, Text, View} from 'react-native'
import {Provider} from 'react-redux'

import NavigationProvider from './components/NavigationProvider'
import Welcome from './screens/Welcome'

import store from './redux/store'

export default class App extends React.Component {
  constructor() {
    super()
    this.state = {
      started: false,
    }
  }
  toggleStart() {
    this.setState({started: !this.state.started})
  }
  renderRoot(Component) {
    return (
      <Provider store={store}>
        <Component toggleStart={this.toggleStart.bind(this)} />
      </Provider>
    )
  }
  render() {
    return this.state.started
      ? this.renderRoot.call(this, NavigationProvider)
      : this.renderRoot.call(this, Welcome)
  }
}
