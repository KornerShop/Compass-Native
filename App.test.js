import React from 'react';
import {shallow} from 'enzyme';

import App from './App';

it('renders properly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
