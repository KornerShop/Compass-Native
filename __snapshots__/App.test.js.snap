// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Main) />
</Provider>
`;
